src/widget/wsingletoncontainer.cpp

```diff
@@ -0,0 +1,61 @@
+#include "widget/wsingletoncontainer.h"
+
+#include <QLayout>
+
+#include "util/assert.h"
+#include "skin/skincontext.h"
+#include "widget/wlibrary.h"
+
+
+WSingletonContainer::WSingletonContainer(QWidget* pParent)
+        : WWidgetGroup(pParent), m_pWidget(NULL), m_pLayout(NULL) { }
+
+void WSingletonContainer::setup(QDomNode node, const SkinContext& context) {
+    setContentsMargins(0, 0, 0, 0);
+    m_pLayout = new QVBoxLayout();
+    m_pLayout->setSpacing(0);
+    m_pLayout->setContentsMargins(0, 0, 0, 0);
+    setLayout(m_pLayout);
+
+    QDomElement containerNode = node.toElement();
+    if (!context.hasNode(node, "ObjectName")) {
+        SKIN_WARNING(node, context)
+                << "Need objectName attribute for Singleton tag";
+        return;
+    }
+    QString objectName = context.selectString(node, "ObjectName");
```

Comment:
Not related to this PR, but it looks like (not tested) that selectString() returns any other sting if "ObjectName" is not found. 
If we change it to returning an empty string in that case we can combine this selectString and hasNode to save on XML loop.